test/pageobjects/mainPage.js

```diff
@@ -1,30 +1,32 @@
-const { config } = require('../../wdio.conf');
 
 class MainPage {
 
-    constructor() {
-        this.url = config.baseUrl;
-        this.consentButtonSelector = '#privacy-policy-banner .banner-button.policy-accept'; 
+    get consentButton() {
+        return $('//div[contains(@class, "policy-accept")]');
     }
 
-    async open() {
-        try {
-            await browser.url(this.url);
-        } catch (error) {
-            console.error('Error opening the page:', error);
-        }
+    get searchResultPlaceholder() {
+        return $('input[placeholder="Search your Address, City or Zip Code"]');
     }
 
-    async consentDataUsage() {
-        await browser.waitForDisplayed(this.consentButtonSelector, {
-            timeout: 5000,
-            timeoutMsg: 'Consent button was not displayed in time'
-        });
-        const isConsentButtonVisible = await browser.isDisplayed(this.consentButtonSelector);
-        if (isConsentButtonVisible) {
-            await browser.click(this.consentButtonSelector);
-        }
+    get firstSearchResult() {
+        return $('div.results-container > div.search-bar-result:first-child');
+    }
+
+    async acceptConsentButton() {
+        await this.consentButton.waitForExist();
+        await this.consentButton.click();
+    }
+
+    async searchCity() {
+        await this.searchResultPlaceholder.click();
+        await this.searchResultPlaceholder.setValue('New York');
```

Comment:
would be better to create this function with parameters and keep your test data 'New York' separately from test

Reply:
Resolved